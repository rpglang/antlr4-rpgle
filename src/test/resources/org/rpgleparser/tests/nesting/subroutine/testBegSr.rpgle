       DCL-S X Int(10);
       BEGSR SomeSubRoutine;
         X = 1234;
       ENDSR;