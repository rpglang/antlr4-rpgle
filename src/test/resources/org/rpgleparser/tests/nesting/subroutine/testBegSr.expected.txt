/*===TOKENS===*/
DS_Standalone            <DCL-S>
ID                       <X>
INT                      <Int>
'('                      <(>
NUMBER                   <10>
')'                      <)>
FREE_SEMI                <;>
OP_BEGSR                 <BEGSR>
ID                       <SomeSubRoutine>
FREE_SEMI                <;>
ID                       <X>
'='                      <=>
NUMBER                   <1234>
FREE_SEMI                <;>
OP_ENDSR                 <ENDSR>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (statement
    (dspec
      DCL-S
      (identifier (free_identifier (idOrKeyword X)))
      (datatype (datatypeName Int) (args ( (expression (number 10)) )))
      ;
    )
  )
  (subroutine
    (begsr
      (freeBEGSR BEGSR (identifier (free_identifier (idOrKeyword SomeSubRoutine))) ;)
    )
    (statement
      (free
        (baseExpression
          (op
            (op_eval
              (evalExpression
                (assignmentExpression
                  (simpleExpression (identifier (free_identifier (idOrKeyword X))))
                  =
                  (expression (number 1234))
                )
              )
            )
          )
        )
        ;
      )
    )
    (endsr (freeENDSR ENDSR ;))
  )
)
/*======*/